core/src/test/java/hudson/model/MockItem.java

```diff
@@ -67,4 +67,41 @@ void enter(Queue q) {
     boolean leave(Queue q) {
         return true;
     }
+
+    private static class MockQueueTask implements Queue.Task {
+        @Override
+        public String getDisplayName() {
+            return null;
+        }
+
+        @Override
+        public String getName() {
+            return null;
+        }
+
+        @Override
+        public String getFullDisplayName() {
+            return null;
+        }
+
+        @Override
+        public void checkAbortPermission() {
+
+        }
+
+        @Override
+        public boolean hasAbortPermission() {
+            return false;
+        }
```

Comment:
BTW these two could just get `default` implementations in `Task`. Normally the impl class is picking up defaults from something else like `Job`, but it would be useful here.